File: full_case_study_solns/_d_mass/simulink/hw10/massParamHW10.m
% sample time for controller
P.Ts = 0.01;

% dirty derivative gain
P.sigma = 0.05;
P.beta = (2*P.sigma-P.Ts)/(2*P.sigma+P.Ts); % dirty derivative gain

% input constraint
P.Fmax = 2;

% tuning parameters
%tr = 5;   % first part
tr = 2.5;  % tuned value of the rise time
zeta = 0.707;
P.ki = 1.5; % integrator gain

% PD gains
% open loop char polynomial and poles
Delta_ol = [1,P.b/P.m,0];
wn = 2.2/tr;
Delta_cl_d = [1,2*zeta*wn,wn^2];
P.kp =  P.m*(Delta_cl_d(3)-Delta_ol(3));
P.kd = P.m*(Delta_cl_d(2)-Delta_ol(2));

File: full_case_study_solns/_d_mass/python/hw10/massParamHW10.py
# Single link arm Parameter File
import numpy as np
# import control as cnt
import sys
sys.path.append('..')  # add parent directory
import massParam as P

Ts = P.Ts  # sample rate of the controller
beta = P.beta  # dirty derivative gain
tau_max = P.F_max  # limit on control signal

#  tuning parameters
tr = 5
# tuned value of the rise time
tr = 2.0
zeta = 0.707
ki = 1.5  # integrator gain


# compute PD gains
# open loop char polynomial and poles
a1 = P.b/P.m
a0 = P.k/P.m

wn = 2.2/tr
alpha1 = 2.0*zeta*wn
alpha0 = wn**2

kp = P.m*(alpha0-a0)
kd = P.m*(alpha1-a1)

print('kp: ', kp)
print('ki: ', ki)
print('kd: ', kd)


